scripts/export_json.py

```diff
@@ -0,0 +1,247 @@
+
+
+import json
+import logging
+import argparse
+import magic
+import binaryninja as binja
+
+LOGNAME = 'binary.log'
+log = logging.getLogger(LOGNAME)
+
+_TYPE_MAP= {
+  'int8_t'  : 'b',
+  'uint8_t' : 'B',
+  'int16_t' : 'h',
+  'uint16_t': 'H',
+  'int32_t' : 'i',
+  'uint32_t': 'I',
+  'long'    : 'l',
+  'ulong'   : 'L',
+  'void'    : 'v',
+  'const* void' : '*v',
+  'void* const' : '*v'
+}
+
+# Utility functions
+def read_bytes_slowly(bv, offset, length):
+  br = binja.BinaryReader(bv)
+  bytestr = []
+  for x in xrange(offset, offset+length):
+    br.seek(x)
+    bytestr.append('{:x}'.format(br.read8()))
+  return ''.join(bytestr)
+
+# Utility wrappers
+def address(addr):
+  return addr
+
+class XRef(object):
+  IMMEDIATE = 0
+  MEMORY = 1
+  DISPLACEMENT = 2
+  CONTROLFLOW = 3
+
+  def __init__(self, addr, reftype, mask=0):
+    self.addr = addr
+    self.type = reftype
+    self.mask = mask
+
+  def __hash__(self):
+    return hash((self.addr, self.type))
+
+class BasicBlock(object):
+  def __init__(self, bv, func, bb):
+    self._bv = bv
+    self._func = func
+    self._bb = bb
+    self.get_xrefs()
+
+  def get_xrefs(self):
+    instrs = list(self._bb.disassembly_text)
+    for insn in instrs:
+      il_insn = self._func.get_lifted_il_at(insn.address)
+
+  def get_bytes_raw(self):
+    offset = self._bb.start
+    length = self._bb.end - self._bb.start
+    bytes = read_bytes_slowly(self._bv, offset, length)
+    return bytes
+
+class Function(object):
+  def __init__(self, bv, func):
+    self._bv = bv
+    self._func = func
+    self._cc = func.calling_convention
+    self._arch = func.calling_convention.arch
+    self._json_object = {}
+    self._json_object['name'] = func.name
+    self._json_object['address'] = address(func.start)
+    self._json_object['parameters'] = self.get_params()
+    self._json_object['return_address'] = self.get_return_address()
+    self._json_object['return_values'] = self.get_return_registers()
+
+  def get_params(self):
+    param_list = []
+    index = 0
+    param_regs = self._cc.int_arg_regs
+    for var in self._func.parameter_vars:
+      param = {}
+      param['name'] = str(var)
+      param['type'] = _TYPE_MAP[str(var.type)]
```

Comment:
What if `str(var.type)` isn't in the type map? This seems like it could be very likely. Perhaps use your code that translates binja types to anvill `Type`s, then use a method on that class to serialize to a string.